crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -295,6 +298,39 @@ impl PublishFlakeRef {
             PublishFlakeRef::Https(https_ref) => FlakeRef::GitHttps(https_ref),
         }
     }
+
+    #[allow(unused)]
+    fn from_github_ref(
+        GitServiceRef {
+            owner,
+            repo,
+            attributes,
+            ..
+        }: GitServiceRef<service::Github>,
+    ) -> Result<Self, ConvertFlakeRefError> {
+        let host = attributes.host.unwrap_or("github.com".to_string());
+
+        let url_str = format!("https://{host}/{owner}/{repo}");
+        let url = WrappedUrl::from_str(&url_str)
+            .map_err(|e| ConvertFlakeRefError::InvalidResultUrl(url_str, e))?;
+
+        let only_rev = attributes.rev.is_some() && attributes.reference.is_none();
+
+        let git_attributes = GitAttributes {
+            rev: attributes.rev,
+            reference: attributes.reference,
+            dir: attributes.dir,
+            all_refs: only_rev.then_some(true),
+            ..Default::default()
+        };
+
+        let git_ref = GitRef {
+            url,
+            attributes: git_attributes,
+        };
+
+        Ok(Self::Https(git_ref))
```

Comment:
Doesn't analyze still need to treat this as `github` so `github` URLs end up in `catalog.json`?

Reply:
I also see an argument in having "normalized" flakerefs in the catalog (i.e. only git urls).
Anyway, that's beyond the scope of this PR, we can refactore the `Publish` interfaces accordingly after landing this and #170